Sources/FirebaseFirestore/FirestoreErrorCode.swift

```diff
@@ -19,33 +39,42 @@ public struct FirestoreErrorCode: Error {
+  public static var invalidArgument: Self { .init(firebase.firestore.kErrorInvalidArgument) }
+  public static var deadlineExceeded: Self { .init(firebase.firestore.kErrorDeadlineExceeded) }
+  public static var notFound: Self { .init(firebase.firestore.kErrorNotFound) }
+  public static var alreadyExists: Self { .init(firebase.firestore.kErrorAlreadyExists) }
+  public static var permissionDenied: Self { .init(firebase.firestore.kErrorPermissionDenied) }
+  public static var resourceExhausted: Self { .init(firebase.firestore.kErrorResourceExhausted) }
+  public static var failedPrecondition: Self { .init(firebase.firestore.kErrorFailedPrecondition) }
+  public static var aborted: Self { .init(firebase.firestore.kErrorAborted) }
+  public static var outOfRange: Self { .init(firebase.firestore.kErrorOutOfRange) }
+  public static var unimplemented: Self { .init(firebase.firestore.kErrorUnimplemented) }
+  public static var `internal`: Self { .init(firebase.firestore.kErrorInternal) }
+  public static var unavailable: Self { .init(firebase.firestore.kErrorUnavailable) }
+  public static var dataLoss: Self { .init(firebase.firestore.kErrorDataLoss) }
+  public static var unauthenticated: Self { .init(firebase.firestore.kErrorUnauthenticated) }
 }
 
 extension FirestoreErrorCode: Equatable {}
 
-// Unfortunately `Error` is not defined as a `enum class` so we need to add
-// these wrappers.
-extension FirestoreErrorCode.Code {
-  public static var ok: Self { firebase.firestore.kErrorOk }
-  public static var none: Self { firebase.firestore.kErrorNone }
-  public static var cancelled: Self { firebase.firestore.kErrorCancelled }
-  public static var unknown: Self { firebase.firestore.kErrorUnknown }
-  public static var invalidArgument: Self { firebase.firestore.kErrorInvalidArgument }
-  public static var deadlineExceeded: Self { firebase.firestore.kErrorDeadlineExceeded }
-  public static var notFound: Self { firebase.firestore.kErrorNotFound }
-  public static var alreadyExists: Self { firebase.firestore.kErrorAlreadyExists }
-  public static var permissionDenied: Self { firebase.firestore.kErrorPermissionDenied }
-  public static var resourceExhausted: Self { firebase.firestore.kErrorResourceExhausted }
-  public static var failedPrecondition: Self { firebase.firestore.kErrorFailedPrecondition }
-  public static var aborted: Self { firebase.firestore.kErrorAborted }
-  public static var outOfRange: Self { firebase.firestore.kErrorOutOfRange }
-  public static var unimplemented: Self { firebase.firestore.kErrorUnimplemented }
-  public static var `internal`: Self { firebase.firestore.kErrorInternal }
-  public static var unavailable: Self { firebase.firestore.kErrorUnavailable }
-  public static var dataLoss: Self { firebase.firestore.kErrorDataLoss }
-  public static var unauthenticated: Self { firebase.firestore.kErrorUnauthenticated }
+extension FirestoreErrorCode {
+  public typealias Code = FirestoreErrorCode
```

Comment:
Similar to `AuthError` - is this in the reference implementation?

Reply:
If you mean in the Obj C API, then yes. `FirestoreErrorCode.Code` is a thing.

Reply:
Ah, okay. I was trying to figure out if this is something that you are doing for some stylistic reason or to match. I think that a comment might be nice to x-ref the shape.

Reply:
Done